src/vcpkg/paragraphs.cpp

```diff
@@ -385,61 +385,85 @@ namespace vcpkg::Paragraphs
 
 namespace
 {
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_any_manifest_text(
+    ExpectedL<SourceControlFileAndLocation> try_load_any_manifest_text(
         StringView text,
         StringView origin,
+        StringView spdx_location,
         MessageSink& warning_sink,
         ParseExpected<SourceControlFile> (*do_parse)(StringView, const Json::Object&, MessageSink&))
     {
         StatsTimer timer(g_load_ports_stats);
-        return Json::parse_object(text, origin).then([&](Json::Object&& object) {
-            return do_parse(origin, std::move(object), warning_sink).map_error(ToLocalizedString);
-        });
+        return Json::parse_object(text, origin)
```

Comment:
If this looks uglier it gets prettier again after https://github.com/microsoft/vcpkg-tool/pull/1240